archive/tar.go

```diff
@@ -0,0 +1,133 @@
+package archive
+
+// special thanks to this medium article:
+// https://medium.com/@skdomino/taring-untaring-files-in-go-6b07cf56bc07
+
+import (
+	"archive/tar"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type tarArchive struct{}
+
+// NewTarArchive creates an Archive that uses the .tar file format.
+func NewTarArchive() Archive {
```

Comment:
if you put this in your own package, you could rename to `New()` since tar would be implied by the package name, ie `tar.New()`